ext/zip/php_zip.stub.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/** @return resource|int|false */
+function zip_open(string $filename) {}
+
+/**
+ * @param resource $zip
+ */
+function zip_close($zip): bool {}
```

Comment:
This return type should actually be `void`.

Reply:
Am I misunderstanding what happens in https://github.com/php/php-src/blob/de4e353c50ee62afe7c2ba5143b1349f64d307bc/ext/zip/php_zip.c#L1108 ?

Reply:
zend_fetch_resource throws an exception (some return is necessary to stop the function from executing further), so the false return value never makes it to userland.

Reply:
thanks for the clarification.

Reply:
We should change those RETURN_FALSE to `return`. This is just a leftover from before I made this function throw. (Same for RETURN_FALSE on zend_parse_parameters failures.)

Reply:
I already though about introducing some macro which makes it clearer that this is an "exception return". Not sure about the name. A simple `return` would still be an improvement.